rsts/concepts/lifecycle_workflow_with_plugin.rst

```diff
@@ -0,0 +1,252 @@
+.. _lifecycle-workflow:
+
+#################################################################
+Understand the Lifecycle of a Flyte Workflow
+#################################################################
+
+After reading this entry you should be able to understand how Flyte's Plugin machinery
+work under the hood and how information flows from the different
+components in Flyte.
+
+Under the hood Flyte relies on a primitive called “Plugins”. Every task
+that you run on Flyte is somehow powered by a plugin. Some of those plugins are native and guaranteed by Flyte system. These native plugins are for example powering your python functions (i.e: Tasks) by running them inside a k8s pod. There are Three native plugins : ``Container``, ``K8sPod`` and ``Sql``.
+
+Addionally there are also plugins which are actual extensions, they create additional infrastructure, talk to other SaaS on your behalf. Examples of those are the Spark or AWS Athena plugins.
+
+A plugin requires code to live in multiple locations
+
```

Comment:
There are plugins which are actual extensions; they create additional infrastructure, and communicate with SaaS on your behalf. Example: Spark, AWS Athena.

 A plugin requires code to live in multiple locations

Reply:
Can we add links to Spark and AWS Athena plugin?